model/project_parser.go

```diff
@@ -508,50 +508,17 @@ func LoadProjectForVersion(v *Version, id string, shouldSave bool) (ProjectInfo,
 			return ProjectInfo{}, errors.Wrap(err, "finding project config")
 		}
 	}
-	// if parser project config number is old then we should default to legacy
-	if pp != nil && pp.ConfigUpdateNumber >= v.ConfigUpdateNumber {
-		if pp.Functions == nil {
-			pp.Functions = map[string]*YAMLCommandSet{}
-		}
-		pp.Identifier = utility.ToStringPtr(id)
-		var p *Project
-		p, err = TranslateProject(pp)
-		return ProjectInfo{
-			Project:             p,
-			IntermediateProject: pp,
-			Config:              pc,
-		}, err
-	}
-
-	if v.Config == "" {
-		return ProjectInfo{}, errors.New("version has no config")
-	}
-	p := &Project{}
-	// opts empty because project yaml with `include` will not hit this case
-	ctx := context.Background()
-	pp, err = LoadProjectInto(ctx, []byte(v.Config), nil, id, p)
-	if err != nil {
-		return ProjectInfo{}, errors.Wrap(err, "loading project")
+	if pp.Functions == nil {
```

Comment:
I'm not sure why this isn't just part of TranslateProject; maybe because we wanted to upsert it correctly? does it seem possible to move it in? (or into findOne)

Reply:
Not sure why it was there either - there are two instances where we do that - one here and one in the agent route that retrieves the parser project. Since both of them don't use TranslateProject I think it's ok to move it into `ParserProjectFindOneById`